src/odemis/driver/orsay.py

```diff
@@ -1021,86 +935,76 @@ def _updatePosition(self, parameter=None, attributeName="Actual"):
-        """
-        if parameter is None:
-            parameter = self._reservoirPar
-        if parameter is not self._reservoirPar:
-            raise ValueError("Incorrect parameter passed to _updateInjectingGas. Parameter should be "
-                             "datamodel.HybridGIS.ReservoirState. Parameter passed is %s." % parameter.Name)
-        if not attributeName == "Actual":
-            return
-        logging.debug("Gas flow is now %s." % self._reservoirPar.Actual)
-        new_value = self._reservoirPar.Actual == STR_OPEN
-        self.injectingGas._value = new_value  # to not call the setter
-        self.injectingGas.notify(new_value)
+        if self._reservoirPar.Actual == self._reservoirPar.Target:
+            logging.debug("Target injectingGas position reached.")
+            self._injectingGasPositionReached.set()
+        else:
+            self._injectingGasPositionReached.clear()
 
-    def _setInjectingGas(self, goal):
+    def _doMove(self, goal):
         """
-        goal (bool): the goal state of the gas flow: (True: "open", False: "closed")
-        return (bool): the new state the gas flow: (True: "open", False: "closed")
+        goal (dict): the goal state of the GIS position and gas flow:
+            {"arm": True (engaged) / False (parked),
+             "injectingGas": True (open) / False (closed)}
 
-        Opens the GIS reservoir if argument goal is True. Closes it otherwise.
-        Also closes the reservoir if the position of the GIS is not engaged.
+        Moves the GIS to working position if argument goal["arm"] is True. Moves it to parking position otherwise.
+        Opens the gas reservoir of the GIS if goal["injectingGas"] is True. Closes it otherwise.
         """
-        if not self.position.value["arm"]:
-            logging.warning("Gas flow was attempted to be changed while not in working position.")
-            goal = False
-        if goal:
-            logging.debug("Starting gas flow.")
-            self._reservoirPar.Target = STR_OPEN
-        else:
-            logging.debug("Stopping gas flow.")
-            self._reservoirPar.Target = STR_CLOSED
-        return self._reservoirPar.Target == STR_OPEN
+        try:
+            if not goal["arm"] == self.position.value["arm"]:  # if the arm needs to move
+                if self.position.value["injectingGas"]:
+                    logging.warning("Moving GIS while gas flow is on.")
+                self._armPositionReached.clear()  # to assure it waits
+                if goal["arm"]:
+                    logging.debug("Moving GIS to working position.")
+                    self._positionPar.Target = STR_WORK
+                else:
+                    logging.debug("Moving GIS to parking position.")
+                    self._positionPar.Target = STR_PARK
+        except KeyError:  # in case "arm" is nog present in goal
+            pass
```

Comment:
Avoid using exceptions for things are happen normally. One of the problem is that there might be other part of your code that fails due to KeyError, and this will hidden now.
Better just write `if "arm" in goal`...